frontend/src/metabase/components/Button.jsx

```diff
@@ -0,0 +1,35 @@
+import React, { Component, PropTypes } from "react";
+
+import Icon from "metabase/components/Icon.jsx";
+
+import cx from "classnames";
+
+import _ from "underscore";
+
+const BUTTON_VARIANTS = [
+    "small",
+    "medium",
+    "large",
+    "primary",
+    "warning",
+    "cancel",
+    "purple",
+    "borderless"
+];
+
+const Button = ({ className, icon, children, ...props }) => {
+    let variantClasses = BUTTON_VARIANTS.filter(variant => props[variant]).map(variant => "Button--" + variant);
```

Comment:
Are you suggesting validating combinations of those props? I'm not sure that's necessary, we certainly don't do anything like that for the CSS classes.

I'll add the basic proptypes though.